kubetest/gcedind/gce_dind.go

```diff
@@ -0,0 +1,622 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This file implements a Kubernetes kubetest deployer based on the scripts
+// in the Mirantis/kubeadm-dind-cluster github repo. These scripts are used
+// to create a containerized, Docker-in-Docker (DinD) based Kubernetes cluster
+// that is running in a remote GCE instance.
+
+package gcedind
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"k8s.io/test-infra/kubetest/process"
+)
+
+var (
+	// gce-dind specific test config flags.
+	gceDinDScriptsPath = flag.String("gce-dind-scripts-path", "",
+		"(gce-dind only) Path to a local clone of Mirantis/kubeadm-dind-cluster scripts. Must be set for gce-dind.")
+	gceDinDImage = flag.String("gce-dind-image", "",
+		"(gce-dind only) A pre-built kubeadm-dind-cluster docker image for node containers. If unset, it will be built from latest github.com://Mirantis/kubeadm-dind-cluster.")
+	gceDinDBuildKubeadm = flag.Bool("gce-dind-build-kubeadm", false,
+		"(gce-dind only) Indicates whether kubeadm should be built.")
+	gceDinDBuildHyperkube = flag.Bool("gce-dind-build-hyperkube", false,
+		"(gce-dind only) Indicates whether hyperkube should be built.")
+	gceDinDIPv6Only = flag.Bool("gce-dind-ipv6-only", false,
+		"(gce-dind only) Indicates whether cluster should run in IPv6-only mode.")
+	gceDinDUpTimeout = flag.Duration("gce-dind-up-timeout", 45*time.Minute,
+		"(gce-dind only) Time limit between starting a cluster and making a successful call to the Kubernetes API.")
+	gceDinDNumNodes = flag.Int("gce-dind-num-nodes", 4,
```

Comment:
Again, let's try to avoid flag explosion.

Either use the existing gcp-nodes flag, or [better yet] make a --num-nodes flag. It can default to 4, and [if set] deployment specific flag variables can take their value from this.

Reply:
Created a common --num-nodes flag.